PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'account.move.line'
-    _inherit = "hr.expense"
-    _inherit = "hr.expense.sheet"
-    _inherit = 'project.project'

Total Changes: 12

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/project_sale_expense/models/account_move_line.py
+++ b/addons/project_sale_expense/models/account_move_line.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class AccountMoveLine(models.Model):
-    _inherit = 'account.move.line'
+    _inherit = ['account.move.line']
 
     def _sale_determine_order(self):
         """ For move lines created from expense, we override the normal behavior.

--- a/addons/project_sale_expense/models/hr_expense.py
+++ b/addons/project_sale_expense/models/hr_expense.py
@@ -3,8 +3,8 @@
 from odoo import api, models
 
 
-class Expense(models.Model):
-    _inherit = "hr.expense"
+class HrExpense(models.Model):
+    _inherit = ["hr.expense"]
 
     @api.depends('sale_order_id')
     def _compute_analytic_distribution(self):

--- a/addons/project_sale_expense/models/hr_expense_sheet.py
+++ b/addons/project_sale_expense/models/hr_expense_sheet.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class HrExpenseSheet(models.Model):
-    _inherit = "hr.expense.sheet"
+    _inherit = ["hr.expense.sheet"]
 
     def _do_create_moves(self):
         """ When creating the move of the expense, if the AA is given in the project of the SO, we take it as reference in the distribution.

--- a/addons/project_sale_expense/models/project_project.py
+++ b/addons/project_sale_expense/models/project_project.py
@@ -7,8 +7,8 @@ from odoo import models, fields
 from collections import defaultdict
 
 
-class Project(models.Model):
-    _inherit = 'project.project'
+class ProjectProject(models.Model):
+    _inherit = ['project.project']
 
     def _get_expenses_profitability_items(self, with_action=True):
         expenses_read_group = self.env['hr.expense']._read_group(
